compiler/parser/RdrHsSyn.hs

```diff
@@ -3084,6 +3085,80 @@ splitHsAppTysPs t = go t []
     go (L _ (HsAppTy _ f a)) as = go f (a : as)
     go f                     as = (f, as)
 
+------------
+
+-- | Desugar a try–catch expressions for exception-handling.
+
+mkTryCatchExpr
+  :: LHsExpr GhcPs
+  -> Located ([AddAnn],[LMatch GhcPs (LHsExpr GhcPs)])
+  -> P (HsExpr GhcPs)
+mkTryCatchExpr tryExpr@(L tryLoc _) (L catchLoc (_, rawAlts)) = do
+
+  let tryCatchVar = mkQualVar $ mkVarOcc "_tryCatch"
+      fromAnyExceptionVar = mkQualVar $ mkVarOcc "fromAnyException"
+
+      wildCard :: LPat GhcPs
+      wildCard = noLoc (WildPat noExt)
+
+      emptyBinds :: LHsLocalBinds GhcPs
+      emptyBinds = noLoc (EmptyLocalBinds noExt)
+
+      tryMatch :: LMatch GhcPs (LHsExpr GhcPs)
+      tryMatch = noLoc
+          $ Match noExt LambdaExpr [wildCard] Nothing
+          $ GRHSs noExt [noLoc $ GRHS noExt [] tryExpr] emptyBinds
+      tryLambda = L tryLoc $ HsLam noExt $ MG noExt (noLoc [tryMatch]) Generated
+
+      -- We need to convert the patterns in catch expression from
+      --    "PATTERN -> EXPR"
+      -- into
+      --    "(fromAnyException -> Some PATTERN) -> Some EXPR".
+      -- The following bindings achieve this.
+
+      report :: SrcSpan -> String -> P a
+      report loc e = addFatalError loc (text e)
+
+      convertLPat :: LPat GhcPs -> P (LPat GhcPs)
+      -- TODO https://github.com/digital-asset/daml/issues/8020
+      --   Emit helpful error for single variable or wildcard pattern
+      --   without type annotation.
```

Comment:
I think I’m not quite convinced we should do anything here. I think it’s reasonable to have the RHS infer the exception type, that’s how Haskell type inference works in general. Special casing this on case in a rather crude way (e.g., I can probably add parentheses to silence the warning) doesn’t seem like a good idea.

Reply:
Ok. I'll remove the TODO.